src/main/kotlin/org/rust/lang/core/grammar/rust.bnf

```diff
@@ -366,7 +367,7 @@ private fn_header ::= CONST? UNSAFE? [ EXTERN abi? ] FN IDENTIFIER
 
 fn_params ::= LPAREN !COMMA params COMMA? RPAREN { pin(".*") = 1 }
 private params ::= [ param (COMMA param)* ]
-param ::= pat COLON type_sum
+param ::= pat COLON type
 
 ret_type  ::= ARROW (type | EXCL) { pin = 1 }
```

Comment:
I agree that this is indeed unfortunate, but `ret_type_spec` does not feel better. It trades API implemeter convenience (I don't have to single it out in the `extends` regexp) for API consumer convenience (I have to add `spec` to code working with functions, although it doesn't add any information to `ret_type` part)